invisible_cities/reco/deconv_functions_test.py

```diff
@@ -0,0 +1,179 @@
+import os
+import string
+import random
+import numpy  as np
+import pandas as pd
+
+
+from hypothesis              import given
+from hypothesis              import reproduce_failure
+from hypothesis.strategies   import text
+from hypothesis.strategies   import floats
+from hypothesis.strategies   import tuples
+from hypothesis.strategies   import composite
+from hypothesis.strategies   import lists
+from hypothesis.extra.numpy  import arrays
+
+from .. reco.deconv_functions import cut_and_redistribute_df
+from .. reco.deconv_functions import drop_isolated_sensors
+from .. reco.deconv_functions import interpolate_signal
+from .. reco.deconv_functions import deconvolution_input
+from .. reco.deconv_functions import deconvolve
+from .. reco.deconv_functions import richardson_lucy
+from .. reco.deconv_functions import InterpolationMethod
+
+from .. core.core_functions   import in_range
+from .. core.core_functions   import shift_to_bin_centers
+from .. core.testing_utils    import assert_dataframes_close
+
+from ..   io.dst_io           import load_dst
+
+from scipy.stats             import multivariate_normal
+
+characters = string.ascii_letters
+
+@composite
+def dst(draw, dimension=[3,10]):
+    columns = draw(lists(text(characters, min_size=5), min_size=dimension[0], max_size=dimension[1], unique=True))
+    values  = draw(lists(arrays(float, 100, floats(1, 1e3, allow_nan=False, allow_infinity=False)), min_size=len(columns), max_size=len(columns)))
+
+    return pd.DataFrame({k:v for k, v in zip(columns, values)})
+
+
+@given(dst())
+def test_cut_and_redistribute_df(df):
+    cut_var       = random.choice (df.columns)
+    redist_var    = random.choices(df.columns, k=3)
```

Comment:
random stuff in a hypothesis-based test is not accepted. If the test were to fail, hypothesis would try reproduce it, but because it's random it will not be able to. Some options are:
- Fix the columns on which you perform the operations. After all, these are all hypothesis-generated numbers, so you can't have any control over them and it doesn't matter which columns you pick.
- Accept also an integer (or many) in the adequate range(s) to select the columns
- Doing the same, but creating a new composite that does that
- Use a random generator from hypothesis (not the most elegant in my opinion)